waku/waku_rln_relay/rln_relay.nim

```diff
@@ -358,6 +378,12 @@ proc generateRlnValidator*(
       payload = string.fromBytes(message.payload)
     case validationRes
     of Valid:
+      let proofMetadataRes = msgProof.extractMetadata()
+      if proofMetadataRes.isErr():
+        # dirty should never happen
+        return pubsub.ValidationResult.Reject
+      # insert the message to the log (never errors)
+      discard wakuRlnRelay.updateLog(msgProof.epoch, proofMetadataRes.get())
```

Comment:
the key here is that we shouldn't update the nullifierlog if the message is wrong (eg proof is invalid). otherwise anyone can "DoS" the nullifieerLog, forcing cleanups and making it useless.